db/seeds.rb

```diff
@@ -1,12 +1,29 @@
 User.create([
   {
-    name: 'test_user3',
-    email: 'test_user3@email.com',
+    name: 'Ginnie',
+    email: 'ginniehench@gmail.com',
     password: 'secret!'
   },
   {
-    name: 'Ginnie3',
-    email: 'ginniehench3@gmail.com',
+    name: 'Ellie',
+    email: 'ellie@awesomedogs.com',
     password: 'secret!'
-  }
+  },
+])
+
+Submission.create([
+  {
+    user_id: 1,
+    text: "Plants don’t get enough credit. They move. You know this. Your houseplant salutes the sun each morning. At night, it returns to center."
+  },
+  {
+    user_id: 1,
+    url: "https://www.nytimes.com/2018/02/02/science/plants-consciousness-anesthesia.html?rref=collection%2Fsectioncollection%2Fscience",
+    title: "Sedate a Plant, and It Seems to Lose Consciousness. Is It Conscious?"
+  },
+  {
+    user_id: 2,
+    post_id: 1,
```

Comment:
setting the IDs like this is making a bunch of assumptions about the database -- mainly that it's in a clean state where the sequence to generate the IDs is set to zero. Running the seeds multiple times would yield different results.

To address that:

- Use `exists?` to guard against creating multiples of the same record when re-running the seeds
- Set local variables for the users and posts referenced here to ensure consistency no matter what state the database is in

Does all that make sense?

Reply:
I've pushed a commit related to this, so I have some kind of solution but wanted to go ahead post this comment since it outlines my process and asks about switching to FactoryBot for seeding the db.

Yes, implementing local variables makes complete sense and I was definitely assuming a blank slate! 

One other thing to consider is that the unique email constraint on the User model blocks the users from being added (based on what I have in my seeds file currently). If the `users` and `submissions` are not added during the same rake, I don't think using local variables would work to pass a `user` to a `submission`. I could switch to using FactoryBot here because I know how to use sequence to change an email address #, which would get around the unique constraint.

Also, I understand the idea of checking whether a record exists but I'm not sure of how to do it. I tried `User.first.exists?` and ended up getting `User.first.nil?` to return an accurate boolean in rails c.